api/src/routing.js

```diff
@@ -160,7 +160,11 @@ app.use(
           // Explicitly allow the telemetry script setting startupTimes
           `'sha256-B5cfIVb4/wnv2ixHP03bHeMXZDszDL610YG5wdDq/Tc='`,
           // AngularJS and several dependencies require this
-          `'unsafe-eval'`
+          `'unsafe-eval'`,
+          // Allow Enketo onsubmit form attribute
+          // https://github.com/medic/cht-core/issues/6988
+          `'unsafe-hashes'`,
```

Comment:
So, it seems that [Firefox does not even support `unsafe-hashes` yet](https://bugzilla.mozilla.org/show_bug.cgi?id=1343950). This makes me wonder why this even works???  (Maybe since the script is not actually getting executed, and so no inline event handlers are being set, it is enough to just have the hash present in `scriptSrc`?) When I tested this locally I did note that the error does not show up in FF when opening a form even if I remove this line... (But presumably it would still throw the error if I was to somehow trigger the event handler code)

Reply:
> So, it seems that Firefox does not even support unsafe-hashes yet. This makes me wonder why this even works???

I think that them not implementing `unsafe-hashes` means that these hashes apply by default? That's just my throw in the dark about why it's working... 

> what is the process you used to generate the hash for the script?

I just used this website: https://report-uri.com/home/hash and pasted the piece of the script that I wanted to hash, namely: `return false;`.

Reply:
> This makes me wonder why this even works???

I works for you too? Can you please check? Just as a caveat, I had to clear all cache/history/everything to get FF to re-evaluate updated CSP.

Reply:
> It works for you too?

Yes this works for me, no problem. (I did also have to clear everything in FF for it to pick up the changes.)

> and pasted the piece of code that I wanted to hash, namely: `return false;`

Ahh that makes sense!  (Also means that, yes, this fix is much more stable than I originally thought (since there seems little chance that they would need to update that extra complex bit of code... :smile: ).

Reply:
Also, just for the record if anyone is reading this later. I was able to create the same has from Linux using this command:

```.sh
echo -n "return false;" | openssl sha256 -binary | openssl base64
```